tests/parametric/test_128_bit_traceids.py

```diff
@@ -438,12 +438,13 @@ def test_w3c_128_bit_propagation_tid_in_parent(self, test_agent, test_library):
         """
         with test_library:
             with test_library.start_span(name="parent", service="service", resource="resource") as parent:
-                with test_library.start_span(name="child", service="service", parent_id=parent.span_id):
+                with test_library.start_span(name="child", service="service", parent_id=parent.span_id) as child:
                     pass
 
         traces = test_agent.wait_for_num_traces(1, clear=True)
-        parent = find_span_in_traces(traces, Span(name="parent", service="service"))
-        child = find_span_in_traces(traces, Span(name="child", service="service"))
+        trace = find_trace(traces, parent.trace_id)
+        parent = find_span(trace, parent.span_id)
+        child = find_span(trace, child.span_id)
 
         parent_tid = parent["meta"].get("_dd.p.tid")
```

Comment:
This looks like one of the cases we were talking about. The spec says the tag `_dd.p.tid` (and any other `_dd.p.*` propagated tag) should be placed in the first span of every trace chunk sent to the agent (meaning the first span in the payload, not necessarily the first span created or the first span closed). Whether that first span is the parent or the child span is undefined/unpredictable and that relationship should not be used for this test.

Reply:
Yup. The initial goal of this PR is to standardize around retrieving spans by ids using find_trace and find_span. Once this change is passing for all enabled tests, then we can move away from sorting spans by start_time (by default) and/or introduce helpers to capture the chunk root spans / trace level tags.

Reply:
I updated this test. There are a few more tests with the same issue